### ISSUE TITLE
Usability suggestions

### ISSUE THREAD

**Issue by premell** [nickcoutsos/keymap-editor#171]
These are some small nitpicks that i noticed while using the editor. Its mainly for inspiration so just ignore the things you disagree with. Sorry for writing the suggestions in a single issue, but i didnt want to spam the board.

1) &none should have a different color than other keys. 
![image](https://github.com/nickcoutsos/keymap-editor/assets/65544203/244edc09-321a-4e86-b442-3ce6bd4b0fc8)
For example this layer looks quite empty at first glance, but really all of these are behaviors
![Screenshot from 2024-01-10 20-22-44](https://github.com/nickcoutsos/keymap-editor/assets/65544203/4dd1edcb-56da-4847-bf59-ce5f6174c5de)

2) when selecting a behavior that requires more keys, like kp, mo, lt, the field should be focused automatically so you dont have to use your mouse:
[Screencast from 2024-01-10 20-24-11.webm](https://github.com/nickcoutsos/keymap-editor/assets/65544203/dc50b582-f845-4d4e-b613-fc8a0d025e9e)

3) You can use escape, enter and arrows when going through the different key alternatives which is really nice. However when you choosen something you still need to use your mouse to confirm the changes. Instead you should be able to hit enter once to select the key and again in "neutral" to confirm the changes (same as clicking ok). Similarly clicking escape when selecting parameter should take you back to choosing behavior and escape again should close the popup.
[Screencast from 2024-01-10 20-26-17.webm](https://github.com/nickcoutsos/keymap-editor/assets/65544203/d3aa2ea6-d8a1-4b73-8475-0f981f99675b)

4) After you change a key, have it selected (like control in this case)
![image](https://github.com/nickcoutsos/keymap-editor/assets/65544203/04302f36-1b4b-4441-9bc7-3646c31a2d35)
If you have a key selected you can use the arrow keys to move to adjacent keys and enter to start editing it. This would be a really big help as you often want to change multiple keys that are beside each other at once.

EDIT: I just noticed that you cannot see my mouse in the recordings, so they might be hard to follow :(

**Comment by nickcoutsos** [comment]
Hey, thanks for sharing this!

Some of these ideas are planned already (or, at least, they're in the back of my mind nagging me regularly).

I'll just run through quickly and summarize

1. styling for `&none`/`&trans` -- yeah, I think that's pretty reasonable. Really simple idea, makes good sense.
2. auto-focus/select the next bind parameter -- this has been requested before, it's probably the next thing I should work on when I can dedicate my attention to it.
3. confirm binding dialog via <kbd>Enter</kbd> -- I don't know if this would necessarily be appropriate as a global handler from that dialog (not that I personally think its bad to do), but you can actually use <kbd>Tab</kbd> now to move focus to that button and activate it by pressing <kbd>Enter</kbd>/<kbd>Space</kbd> ![Screen Shot 2024-01-10 at 3 26 33 PM](https://github.com/nickcoutsos/keymap-editor/assets/1165066/b5a86dc4-9fe7-4085-a665-d68eb08eb34b)
4. Navigation between layer binds via keyboard -- yowza.

I've thought about that a number of times. Probably at the very least because these elements are interactive they should be replaced with proper `button` elements so that they're capable of receiving focus and being triggered by the keyboard. This would also mean being able to navigate via <kbd>Tab</kbd> _but_ using arrow keys to do it two-dimensionally would be even more helpful. It's tricky, though, "adjacent" has a very obvious meaning in a perfectly ortholinear keyboard but it gets fuzzier when you introduce any kind of stagger, and it's even weirder when keys are rotated either individually or in groups. If you're focused on the middle key of the bottom row of the thumb cluster and you press "up" do you expect to land on:
- the left-most key from the top row of the thumb cluster? It covers the space 1 key unit along the y-axis from your current position
- the middle key from the top row of the thumb cluster? It's closest to the space 1 key unit up from the current position in the "local" space of that rotated key, and treating the layout as an ortholinear grid that would otherwise have been positioned directly above it. This might also be the behavior you'd expect in a keyboard like _a.dux_ with its pinky column splay.

I don't even want to think about combining up/down with left/right to get 8 directions of movement.

That's not to say this feature will never happen, just that I'll have to approach it carefully. Most likely you'll see support for <kbd>Tab</kbd> navigation through the layer binds far sooner.
